### PR TITLE
Docs: new intro, principles, and "block as interface"

### PR DESCRIPTION
**Pull request by mtias**
This is the start of some reorganization of the handbook documents to encompass a more living resource spanning project vision, interface guidelines, all the way down to the API documentation, references, and tutorials.

Kudos to @alexislloyd for the help in getting these started.

A rethinking of the IA of the handbook navigation will also be required separately to give some better hierarchy and clarity to the now quite huge sidebar.

I'm also moving the logo and design resources to the "references" document as it didn't seem worth the front-page.

Let's discuss!

cc @jasmussen @karmatosed @kjellr @chrisvanpatten

### CODE REVIEW

**Comment by kjellr** [comment]
This is great. 👏  I especially like the way it frames Gutenberg for both users and designers/developers in the principles section. 

I think the first part of `blocks.md` could use some smoothing over — It's a little hard to parse currently. But I'll hold off on minute comments until someone has a chance to do a more thorough pass at the copy. 

> I'm also moving the logo and design resources to the "references" document as it didn't seem worth the front-page.

👍 Good move.

**Comment by jasmussen** [comment]
I love it. I think it's such a vast improvement over what's there, I almost think we should get this in wholesale and do corrections after the fact rather than keep this in limbo. It's solid.

I do think the GIF (https://cldup.com/kZXGDcGPMU.gif) could potentially be polished a little bit, showing something a little more exciting, perhaps the slash command and an exciting block, maybe with a better framerate and no mouse clicks visible. But that's a small niggle, and something I can do in a weeks time or so if there are no other takers.

**Comment by chrisvanpatten** [comment]
A more comprehensive "meet Gutenberg" video might be nice in place of the GIF. A video can have an audio track and captions, and can also be streamed instead of downloaded wholesale (#webperf), paused, etc. I think for this case a video is probably the better long term option. _Definitely not a blocker to merge_ to be clear, but I think it would provide a much better UX.

I also think this is a great starting point and worth merging now and iterating later 🚢

File: docs/reference.md [hunk 1]
```diff
@@ -4,3 +4,14 @@
 - [Coding Guidelines](../docs/reference/coding-guidelines.md)
 - [Testing Overview](../docs/reference/testing-overview.md)
 - [Frequently Asked Questions](../docs/reference/faq.md)
+
+## Logo
+<img width="200px" src="https://raw.githubusercontent.com/WordPress/gutenberg/master/docs/final-g-wapuu-black.svg?sanitize=true" alt="Gutenberg Logo" />
```
Comment (chrisvanpatten) [inline docs/reference.md]: The `width` here doesn't need `px`. Most modern browsers will totally ignore text in this field (try setting the width to `200in` and see it still renders at 200px). Width/height attributes are implicitly `px` according to the spec.

**Comment by jasmussen** [comment]
> Yes, agreed. Would you rather leave the gif out for now?

No strong blocking opinions. If anyone has bandwidth to create a better GIF, cool. If not, I'll try and get some bandwidth myself.

**Review by mcsf — APPROVED** [review]

**Comment by gziolo** [comment]
Was it intentionally not included in `manifest` config file to avoid having it exposed in Gutenberg handbook?

**Comment by mtias** [comment]
Indeed, don't want to have to change urls too much.

**Comment by karmatosed** [comment]
Really awesome to see this!
